contrib/generic_proxy/filters/network/test/codecs/dubbo/config_test.cc

```diff
@@ -0,0 +1,460 @@
+#include <memory>
+
+#include "source/extensions/common/dubbo/message_impl.h"
+
+#include "test/extensions/common/dubbo/mocks.h"
+#include "test/mocks/server/factory_context.h"
+
+#include "contrib/generic_proxy/filters/network/source/codecs/dubbo/config.h"
+#include "contrib/generic_proxy/filters/network/test/mocks/codec.h"
+#include "gtest/gtest.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace NetworkFilters {
+namespace GenericProxy {
+namespace Codec {
+namespace Dubbo {
+namespace {
+
+using testing::_;
+using testing::ByMove;
+using testing::Return;
+
+using namespace Common::Dubbo;
+
+inline void addInt32(Buffer::Instance& buffer, uint32_t value) {
+  value = htobe32(value);
+  buffer.add(&value, 4);
+}
+
+inline void addInt64(Buffer::Instance& buffer, uint64_t value) {
+  value = htobe64(value);
+  buffer.add(&value, 8);
+}
```

Comment:
User Buffer::Instance has writeBEInt so you can just use them directly.

Reply:
get it.